src/ac_copilot_trainer/modules/lap_archive.lua

```
@@ -0,0 +1,416 @@
+  local f, ferr = io.open(path, "w")
+  if not f then return false, "open failed: " .. tostring(ferr) end
+  if not f:write(raw) then
+    pcall(function() f:close() end)
+    pcall(os.remove, path)
+    return false, "write failed"
+  end
+  local flushOk, flushRes = pcall(function() return f:flush() end)
+  if not flushOk or flushRes == false then
+    pcall(function() f:close() end)
+    pcall(os.remove, path)
+    local ferr = (not flushOk) and tostring(flushRes) or "flush returned false"
```

Luacheck (1.2.0)

[warning] shadowing definition of variable 'ferr' on line 361

(W421)